### PR TITLE
lineTopic form

### LINKED ISSUE #956: LineHeader form
FE PR: https://github.com/HSLdevcom/jore-map-ui/pull/959
BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/85

Copy the idea from the old UI:
image

### PR DESCRIPTION
**Pull request by Divinare**
#closes #956 

BE PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/85

* edit / save a new lineHeader
* note that even startDate of a lineHeader can be edited (startDate is part of LineTopic primarykey which is lineId + startDate combination)
   * this is why originalStartDate is being sent to backend when saving

### CODE REVIEW

File: src/components/sidebar/lineView/LineInfoTab.tsx [hunk 1]
```diff
@@ -48,16 +49,16 @@ class LineInfoTab extends React.Component<
...
     }
 
     async componentWillMount() {
         const lineId = this.props.lineStore!.line!.id;
-        const lineTopics: ILineTopic[] = await LineTopicService.fetchLineTopicsForLineId(
+        const lineHeaders: ILineHeader[] = await LineHeaderService.fetchLineHeaders(
             lineId
         );
-        this.initLineTopicItems(lineTopics);
+        this.initLineHeaderItems(lineHeaders);
```
Comment (ahjyrkia) [inline src/components/sidebar/lineView/LineInfoTab.tsx]: Maybe unnecessary to make this in to a function of its own? Just moving around lineHeaders and not making changes to it.

File: src/components/sidebar/lineView/LineHeaderTable.tsx [hunk 2]
```diff
@@ -60,42 +72,41 @@ class LineTopicTable extends React.Component<ILineTopicListProps> {
                                 </th>
                                 <th className={s.columnHeader} />
                             </tr>
-                            {this.props.lineTopics.map(
-                                (lineTopic: ILineTopic, index: number) => {
-                                    const isCurrentLineTopic = _.isEqual(
-                                        this.props.currentLineTopic,
-                                        lineTopic
+                            {this.props.lineHeaders.map(
```
Comment (ahjyrkia) [inline src/components/sidebar/lineView/LineHeaderTable.tsx]: Maybe move this to own function to make render-function easier to read. Looks a bit messy right now but the reason for this might also be the current row length limit.

{ this.lineHeaderRows() }

File: src/components/sidebar/lineView/LineHeaderTable.tsx [hunk 3]
```diff
@@ -4,34 +4,46 @@ import _ from 'lodash';
...
+            .toLink();
+
+        navigator.goTo(newLineHeaderLink);
     };
     render() {
         return (
-            <div className={s.lineTopicTableView}>
+            <div className={s.lineHeaderTableView}>
                 <SidebarHeader hideCloseButton={true}>
                     Linjan otsikot
```
Comment (ahjyrkia) [inline src/components/sidebar/lineView/LineHeaderTable.tsx]: Not part of this pr but why is the title indented?

Comment (ahjyrkia) [inline src/components/sidebar/lineView/LineHeaderTable.tsx]: Not part of this pr but why is the title indented?

File: src/components/sidebar/lineView/LineHeaderTable.tsx [hunk 4]
```diff
@@ -60,42 +72,41 @@ class LineTopicTable extends React.Component<ILineTopicListProps> {
...
+                                        this.props.currentLineHeader,
+                                        lineHeader
                                     );
                                     return (
                                         <tr
                                             key={index}
                                             className={
-                                                isCurrentLineTopic
-                                                    ? s.lineTopicRowHighlight
+                                                isCurrentLineHeader
```
Comment (ahjyrkia) [inline src/components/sidebar/lineView/LineHeaderTable.tsx]: Use classnames? 
```
className={classnames(
  s.lineHeaderRow,
  isCurrentLineHeader ? s.lineHeaderRowHighlight : null
)}
```

File: src/components/sidebar/lineView/LineInfoTab.tsx (hunk 1 above)
Comment (Divinare) [reply src/components/sidebar/lineView/LineInfoTab.tsx]: We try to keep react's lifecycle methods as clean as possible. The details of that code is not very interesting so it is better to put it in its own method

File: src/components/sidebar/lineView/LineHeaderTable.tsx (hunk 3 above)
Comment (Divinare) [reply src/components/sidebar/lineView/LineHeaderTable.tsx]: Prettier indents it automatically. We need to change that 100 char per line restriction at some point

File: src/components/sidebar/lineView/LineHeaderTable.tsx (hunk 4 above)
Comment (Divinare) [reply src/components/sidebar/lineView/LineHeaderTable.tsx]: There is no such class as .lineHeaderRow?
